api-docs/interface/server-socket.md

```diff
@@ -4,35 +4,72 @@
 Connect to the websocket at `/connect?gameId={game ID}&playerId={player ID}`
 - Calls handleConnection()
 
-## Connection.ts
-
 ### Data Structs:
 
-<!-- This should move somewhere else (the game object?) -->
-`connections` (`Map<string, Map<string, WebSocket>>`)
-
 - Every message is a JSON object with type as a string, and potentially other fields depending on the type.
 
 - When a client connects to the server and the game has already started, the server immediately sends a question or endQuestion (or lobby if client is the host) message to catch them up to the current state of the game. Before then, the client just displays “Waiting for the host to start the game” or something. This way the same logic handles both “client joined and is waiting for game to start” and “client joined late and is waiting for the server to respond to them.”
 
-## 
+- Leaderboard
+    - `name` (string) : the player's username 
+    - `score` (number) : the score of the player
+
+# Types
+
+These are all defined in `respTypes.ts` with 
+- a class that contains the name and data
+- an interface that contains the data
 
 These messages are the same for players and the host unless specified.
 
 Possible types (all sent from server to client unless otherwise specified):
-end
 
-Other fields: reason (string)
+
+## `end`
+
 Whoever sends this message is closing the connection for the specified reason. E.g. the server could send {“type”: “end”, “reason”: “That game does not exist”}
-startQuestion
 
-Other fields: questionText, answerText (same as in JSON), endTimestamp: numeric UNIX timestamp (milliseconds) when the timer to answer this question will out, score (number): player’s current score
+Fields: 
+- reason (string)
+
+## `beginQuestion`
+
 Sent by server whenever a new question is available
-endQuestion
-Fields: correct (boolean), correctAnswers (numeric array), score (number) = the player’s current score, scoreChange (number) = how much their score increased due to this question, 
-Host only: leaderboard: sorted array of [{name: string, score: number }, …]
+
+Fields: 
+- `questionText`: The text for the question
+- `answerTexts`: The answer texts (same as in JSON)
+- `time` (number): The number of *miliseconds* that are left in the question (where 0 is the end of the question)
+- `score` (number): player’s current score, the host will recieve garbage
+- `username`: the name of the player. 
+future:
+- `questionIndex` : the index of the question started.
```

Comment:
I'd rather have this alongside all the other fields maybe with a note "backend does not send this currently" or something, instead of a "future" section. As frontend is looking for questionIndex right now, not in the future, and isn't going to work without it.